x-pack/plugins/security/public/management/roles/edit_role/privileges/es/index_privilege_form.tsx

```diff
@@ -149,11 +160,30 @@ export class IndexPrivilegeForm extends Component<Props, State> {
     );
   };
 
+  private loadFLSOptions = (indexNames: string[], force = false) => {
+    if (!force && (this.isFieldListLoading || indexNames.length === 0)) return;
+
+    this.isFieldListLoading = true;
+    this.setState({
+      isFieldListLoading: true,
+    });
```

Comment:
Why is this duplicated?

Reply:
We use the field within `state` to control UI behavior (loading spinners and such), and we use field outside of `state` to ensure that we don't have more than one API request in-flight at any one time. See https://github.com/elastic/kibana/pull/98681#discussion_r625564714